crates/turborepo-ui/src/log_replayer.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
+    fmt::Display,
+    fs::File,
+    io::{BufRead, BufReader, Write},
+};
+
+use tracing::{debug, warn};
+use turbopath::AbsoluteSystemPath;
+
+use crate::{Error, StyledObject, UI};
+
+#[allow(dead_code)]
+pub struct PrefixedUI<D, W> {
+    ui: UI,
+    prefix: StyledObject<D>,
+    output: W,
+}
+
+#[allow(dead_code)]
+impl<D: Display + Clone, W: Write> PrefixedUI<D, W> {
+    pub fn new(ui: UI, prefix: StyledObject<D>, output: W) -> Self {
+        Self { ui, prefix, output }
+    }
+
+    pub fn output(&mut self, message: impl Display) -> Result<(), Error> {
```

Comment:
Since this differs from `PrefixUI.Output` in Go we should add a warning comment that it will output the prefix is passed an empty message unlike Go.